### ISSUE TITLE
⛳ Implement and/or Condition Sections for While Loops 

### ISSUE THREAD

**Issue by hapily04** [SkriptLang/Skript#5883]
### Suggestion

The best implementation of and/or condition acceptance has been exemplified [here](https://github.com/SkriptLang/Skript/pull/5165). Goal should be to implement before the 2.7 release as it's so incredibly similar to the changes to if's, and would be silly to release it in the next version if it can be avoided. 
Potential syntax: 
```
[do] while (any|at least one [of])
[do] while [all]
[do] while <.+>
```

### Why?

It matches the changes to if statements and enhances the viability of while loops.

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this suggestion.

**Comment by TheLimeGlass** [comment]
do while is already possible.

**Comment by hapily04** [comment]
> do while is already possible.

You're correct in that **do while** _is_ already possible; however, this issue is not questioning "**do while**"'s existence. This issue is addressing **and/or condition acceptance** within **_while loops_** as was done with **if statements**. The **if statement** enhancement is going to be in _**2.7**_, so that's why the **target release** tag of 2.7 was added to this issue due to its _relativity and willingness_ shown by @Pikachu920 to try to get this done in time (https://discord.com/channels/135877399391764480/836220422223036467/1137927545547935834)

# SHOWCASE

### Current
```hs
while player is op: # I also want this to run if the player is in adventure mode, but I cannot check if their gamemode is adventure OR op in while loops' current state
  heal player
  wait 1 tick
```

### Proposed
```hs
while any:
  player is op
  player's gamemode is adventure
then:
  heal player
  wait 1 tick
```

Adopting this enhancement _proven_ to be useful [here](https://github.com/SkriptLang/Skript/pull/5165) for **if statements** would not only allow the proposed example to work, but also allow users to require **multiple conditions** to be met for the while loop to run in a **neat** fashion.

**Comment by sovdeeth** [comment]
Other possible syntax could be
It could be nice to do 
```
while:
    xyz
do:
   abc
```
and
```
do:
   abc
while:
   xyz
```
The ability to swap the two around might get confusing if you have a few in a segment of code, though.

**Comment by GamingwithNP** [comment]
@Fusezion and @TheMug06, you are right, sorry, I tried finding any of the same post as my idea but I did not find any. And @TheMug06, I see what you are asking as your first post in my suggestion, but I think it should be as @sovdeeth suggested, however I was thinking having it be like `run` or `execute` but `do` also seems good!
